compiler/optimizer/OMRValuePropagation.hpp

```diff
@@ -281,118 +284,32 @@ class ValuePropagation : public TR::Optimization
    ValueConstraint *copyValueConstraints(ValueConstraints &valueConstraints);
 
 #if USE_TREES
-   class ValueConstraintHandler : public TREE_HANDLER<ValueConstraint>
+   class ValueConstraintHandler : public TREE_HANDLER <ValueConstraint>
       {
       public:
-      void setVP(TR::ValuePropagation *vp) {_vp = vp;}
```

Comment:
@andrewcraik @0xdaryl Here are my two cents...

Thanks @0xdaryl. I looked around the source code and the limitations of typedefs don't really apply in this case. However, given that we are trying to set a precedent on how one will map a class to the TR namespace and I can't predict how future classes will be used (i.e. one of the typedef limitations might become relevant), I vote for creating a new VP class in TR namespace in ValuePropagation.hpp. FWIW, I also don't see a clear downside to this option.

I had a quick chat with Vijay about this as well. He votes for the second option, i.e creating a new TR::VP class. His reasoning is that he would like us to keep the number of concepts used to construct an optimization class simple and since we are already using (virtual) inheritance for these opts, we should stick to it for name mapping as well.